pandas/core/window/indexers.py

```diff
@@ -170,3 +170,63 @@ def get_window_bounds(
         end = np.concatenate([end_s, end_e])
 
         return start, end
+
+
+class GroupbyRollingIndexer(BaseIndexer):
+    """Calculate bounds to compute groupby rolling, mimicking df.groupby().rolling()"""
+
+    def __init__(
+        self,
+        index_array: Optional[np.ndarray],
+        window_size: int,
+        groupby_indicies: Dict,
+        rolling_indexer: Union[Type[FixedWindowIndexer], Type[VariableWindowIndexer]],
```

Comment:
should add this to _typing (followon ok)

Reply:
Sure can do in a follow on